provider/database.py

```diff
@@ -67,9 +68,42 @@ def disableTrigger(self, triggerFQN, status_code):
             logging.error('[{}] Uncaught exception while disabling trigger: {}'.format(triggerFQN, e))
 
 
-    def changesFeed(self):
-        return self.database.infinite_changes(include_docs=True)
+    def changesFeed(self, timeout, since = None):
+        if since == None:
+            return self.database.infinite_changes(include_docs=True, heartbeat=(timeout*1000))
+        else:
+            return self.database.infinite_changes(include_docs=True, heartbeat=(timeout*1000), since=since)
+
+    def createCanary(self):
+        retryCount = 3
+
+        while retryCount >= 0:
+            if retryCount == 0:
+                logging.error('[canary] Retried and failed {} times to create a canary'.format(retryCount))
```

Comment:
If canary document is not created by this function will the attempt to create it again occur when the canary is updated?